src/classes/QueryPredicate.cls

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class QueryPredicate {
```

Comment:
I think the benefit is that with the query builder's methods not (previously) being accessible (since they have been hidden behind the SObjectrRepository), we previously were making many highly specific methods in the repository class / interface related to getting things.  with the query predicates, you can encapsulate a whole list of criteria within one single "get" request.

Here's an example:
`QueryPredicate emailInAddresses = QueryPredicate.EQUALS(Schema.Lead.Email, contactEmails);
QueryPredicate notClosed = QueryPredicate.EQUALS(Schema.Lead.IsConverted, false);`
List<SObject> leads = this.leadRepo.get(new List<QueryPredicate> {emailInAddresses, notClosed});